ls1intum/Artemis
# `Users`: Add passwort reset by username for internal users

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


If people change their email address they might not remember which one they need for the password reset if the change isn't propagated to Artemis. As we do not expose which addresses exist in the system and which ones don't for security reasons, we should offer another way to reset the password.

### Description

I used the same field, simplified the form in the frontend and allow the user to enter a string that gets compared to the fields `login` and `email`.

### Steps for Testing

To get an internal account:
Either you participated in #4508, then you can take your Edx account on TS1 or you should take TS2 and register with an email address you have access to but don't have an account on yet.

1. Use the password reset functionality to reset your password. Insert your **email address** into the form field.
2. Check your mails and change your password
3. Make sure that you can't log in with your old password but with your new one
4. Use the password reset functionality to reset your password. Insert your **username** into the form field.
5. Check your mails and change your password
6. Make sure that you can't log in with your old password but with your new one

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
